add check for preferred format of Content Type (#31)
core-wg/cf-reg-update: draft-ietf-core-cf-reg-update.md

normative:
  RFC7120: iana-early
  RFC7252: coap

>   RFC9110:

Reviewer: Suggest aliasing this to `http-sema` and, in the refs,  using that label instead of the raw RFC number.

1. The media type associated with the requested Content-Format must exist (or must have been approved for registration) in the "Media Types" registry {{IANA.media-types}};
1. The optional parameter names must have been defined in association with the media type, and any parameter values associated with such parameter names must be as permitted;
1. The Content Type is in the preferred format defined in {{preferred-format}};
1. If a Content Coding is specified, it must exist (or must have been approved for registration) in the "HTTP Content Coding" registry of the "Hypertext Transfer Protocol (HTTP) Parameters" {{IANA.http-parameters}}.

For the 0-255 range, in addition to the checks described above, the DE is instructed to also evaluate the requested codepoint concerning the limited availability of the 1-byte codepoint space.
For the 256-9999 range and the 10000-64999 range, this criterion does not apply.

<!-- Should these actually use BCP14 MUSTs? -->

## Preferred Format for the Content Type Field {#preferred-format}

This section defines the preferred string format for including a requested Content Type into the CoAP Content-Formats registry.

> During the review procedure, the Designated Expert(s) or IANA may rewrite a requested Content Type to this preferred string format prior to approval.

Reviewer: Maybe

> _"During the review process, the Designated Expert(s) or IANA may rewrite a requested Content Type into this preferred string format before approval."_

> During the review procedure, the Designated Expert(s) or IANA may rewrite a requested Content Type to this preferred string format prior to approval.

Reviewer: ```suggestion
During the review process, the Designated Expert(s) or IANA may rewrite a requested Content Type into this preferred string format before approval.
```
